### PR TITLE
Added ReadFileStreamToBuffer from Mender

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by oleorhagen** [comment]
> Is this worth merging as is or shall we do the polishing right away?

I vote for polishing

**Comment by olehermanse** [comment]
> I vote for polishing

@oleorhagen Easy to say when you're not the one doing it ;)

**Comment by olehermanse** [comment]
@oleorhagen @vpodzime Please review cleanup in last commit.

File: libutils/file_lib.c [hunk 1]
```diff
@@ -73,38 +73,28 @@ Writer *FileRead(const char *filename, size_t max_size, bool *truncated)
...
-            return 1;
+            return FILE_ERROR_READ;
         }
         else if (n == 0)
         {
             break;
         }
         bytes_read += n;
     }
-    if (bytes_read != bytes_expected)
```
Comment (oleorhagen) [inline libutils/file_lib.c]: I guess this is intended, but now the function is not guaranteed to fill the buffer completely anymore, which was the main essence of the function in MBD.

Comment (olehermanse) [reply libutils/file_lib.c]: @oleorhagen technically, no, that was not guaranteed before, but I understand what you mean. Yes, it is intended. We just have to check the return code differently in MBD.

Comment (oleorhagen) [reply libutils/file_lib.c]: I see. I guess it makes sense with the name change also.

Comment (olehermanse) [reply libutils/file_lib.c]:
```C
if (ReadFile(file, bytes_expected + 1) == bytes_expected)
{
    // SUCCESS
}
else
{
    // FAIL
}
```

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4384/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4384/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4384/

File: tests/unit/file_lib_test.c [hunk 2]
```diff
@@ -1614,6 +1614,24 @@ static void test_file_read(void)
...
+static void test_read_file_stream_to_buffer(void)
+{
+    setup_tempfiles();
+
+    {
+        const size_t length = strlen(TEST_STRING);
+        char buf[1024] = {0};
+        FILE *const file = safe_fopen(TEST_FILE, "r");
+        assert_true(file != NULL);
+        const ssize_t read = ReadFileStreamToBuffer(file, length, buf);
```
Comment (vpodzime) [inline tests/unit/file_lib_test.c]: Please name the variable `n_read` or something that is not masking the `read()` function.

File: tests/unit/file_lib_test.c [hunk 3]
```diff
@@ -1614,6 +1614,24 @@ static void test_file_read(void)
...
+        fclose(file);
```
Comment (vpodzime) [inline tests/unit/file_lib_test.c]: Would be also nice to do a read again here to check that it fails correctly on a closed stream.

File: tests/unit/file_lib_test.c [hunk 4]
```diff
@@ -1614,6 +1614,24 @@ static void test_file_read(void)
     return_to_test_dir();
 }
 
+static void test_read_file_stream_to_buffer(void)
+{
+    setup_tempfiles();
+
+    {
```
Comment (vpodzime) [inline tests/unit/file_lib_test.c]: Why this sub-block?

File: tests/unit/file_lib_test.c (hunk 2 above)
Comment (olehermanse) [reply tests/unit/file_lib_test.c]: Thanks, fixed.

File: tests/unit/file_lib_test.c (hunk 3 above)
Comment (olehermanse) [reply tests/unit/file_lib_test.c]: I don't think this is defined behavior, or at least I couldn't find anything about calling fread on a closed file in the C standard, other than:

> The value of a pointer to a FILE object is indeterminate after the associated file is closed.

Maybe not a good idea to test this?

File: tests/unit/file_lib_test.c (hunk 4 above)
Comment (olehermanse) [reply tests/unit/file_lib_test.c]: Copy paste from the other tests, just a way to be able to add multiple test cases with the same variable names reused.

File: tests/unit/file_lib_test.c (hunk 3 above)
Comment (vpodzime) [reply tests/unit/file_lib_test.c]: OK

File: tests/unit/file_lib_test.c (hunk 4 above)
Comment (vpodzime) [reply tests/unit/file_lib_test.c]: OK

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4385/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4385/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4385/

**Pull request merged**